# 🌍 DeMark-World - Google Colab Setup

**Universal AI Video Watermark Removal** - Works with Sora, Runway, Veo, Pika, Kling, Luma and more!

**Important:** Make sure you have GPU enabled in Runtime > Change runtime type > Hardware accelerator > GPU (T4 or better)

## Features:
- 🎯 Universal watermark detection (YOLOv11)
- ⚡ LAMA model for fast removal
- 🎬 E2FGVI_HQ for time-consistent (flicker-free) results
- 🚀 Modern stack (PyTorch 2.5+, NumPy 2.x)

**Repo**: https://github.com/linkedlist771/DeMark-World

---

## Step 1: Install System Dependencies

In [ ]:
# Install FFmpeg (required for video processing)
!apt-get update -qq
!apt-get install -y ffmpeg

# Verify installation
!ffmpeg -version | head -n 1

## Step 2: Clone DeMark-World Repository

In [ ]:
# Clone the repository
!git clone https://github.com/linkedlist771/DeMark-World.git
%cd DeMark-World

## Step 3: Install UV Package Manager

UV is 10-100x faster than pip and handles dependencies perfectly. This is the modern way!

In [ ]:
# Install UV package manager
!pip install -q uv

# Verify installation
!uv --version

## Step 4: Install All Dependencies with UV

This single command installs EVERYTHING - no dependency conflicts! 🎉

In [14]:
# Install all dependencies (this may take 2-3 minutes)
print("Installing all dependencies with UV...")
print("This will install: PyTorch 2.5+, transformers, diffusers, mmcv-full, and more")
print("")

# First, create the UV environment and install build dependencies INTO it
print("Step 1: Setting up UV environment with build dependencies...")
!uv venv --clear
!uv pip install --python .venv/bin/python setuptools hatchling uv-dynamic-versioning editables

# Then install all dependencies with build isolation disabled
print("\nStep 2: Installing all project dependencies...")
!uv sync --no-dev --no-build-isolation

print("\n" + "="*60)
print("✅ INSTALLATION COMPLETE!")
print("="*60)
print("All dependencies installed successfully.")
print("Proceed to Step 5 to verify.")

Installing all dependencies with UV...
This will install: PyTorch 2.5+, transformers, diffusers, mmcv-full, and more

Step 1: Setting up UV environment with build dependencies...
Using CPython 3.12.12 interpreter at: /usr/bin/python3
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
Resolved 12 packages in 57ms
Prepared 1 package in 9ms
Installed 12 packages in 14ms
 + dunamai==1.25.0
 + editables==0.5
 + hatchling==1.28.0
 + jinja2==3.1.6
 + markupsafe==3.0.3
 + packaging==26.0
 + pathspec==1.0.4
 + pluggy==1.6.0
 + setuptools==82.0.0
 + tomlkit==0.14.0
 + trove-classifiers==2026.1.14.14
 + uv-dynamic-versioning==0.13.0

Step 2: Installing all project dependencies...
Resolved 209 packages in 0.99ms
Uninstalled 1 package in 6ms
Installed 38 packages in 71ms
Prepared 153 packages without build isolation in 1m 29s
Uninstalled 9 packages in 30ms
Installed 153 packages in 1.30s
 + addict==2.4.0
 + aiofiles==25.1.0
 + aiosqlite==0.21.0
 + altair==5.5.0
 + annot

## Step 5: Verify Installation

In [15]:
# Activate UV environment and verify packages
import sys
sys.path.insert(0, '/content/DeMark-World/.venv/lib/python3.12/site-packages')

import torch
import numpy as np
import torchvision

print("="*60)
print("ENVIRONMENT VERIFICATION")
print("="*60)
print(f"Python: {sys.version.split()[0]}")
print(f"PyTorch: {torch.__version__}")
print(f"Torchvision: {torchvision.__version__}")
print(f"NumPy: {np.__version__}")
print(f"CUDA available: {torch.cuda.is_available()}")

if torch.cuda.is_available():
    print(f"GPU: {torch.cuda.get_device_name(0)}")

# Check key packages
try:
    import cv2
    print(f"✅ OpenCV: {cv2.__version__}")
except:
    print("❌ OpenCV: Not available")

try:
    import mmcv
    print(f"✅ MMCV: {mmcv.__version__}")
except:
    print("❌ MMCV: Not available (will be installed on first run)")

try:
    from ultralytics import YOLO
    print("✅ Ultralytics: Available")
except:
    print("❌ Ultralytics: Not available")

try:
    import transformers
    print(f"✅ Transformers: {transformers.__version__}")
except:
    print("❌ Transformers: Not available")

try:
    import diffusers
    print(f"✅ Diffusers: {diffusers.__version__}")
except:
    print("❌ Diffusers: Not available")

print("="*60)
print("\n🎉 All checks passed! Ready to process videos.")

ENVIRONMENT VERIFICATION
Python: 3.12.12
PyTorch: 2.9.0+cu126
Torchvision: 0.24.0+cu126
NumPy: 2.0.2
CUDA available: True
GPU: Tesla T4
✅ OpenCV: 4.13.0
✅ MMCV: 1.7.2
Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.


/content/DeMark-World/.venv/lib/python3.12/site-packages/mmcv/__init__.py:20: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  warnings.warn(


✅ Ultralytics: Available
✅ Transformers: 5.0.0
✅ Diffusers: 0.36.0

🎉 All checks passed! Ready to process videos.


## Step 6: Initialize DeMark-World

In [16]:
# Set up environment
%cd /content/DeMark-World
sys.path.insert(0, '/content/DeMark-World/src')

# Create output directories
!mkdir -p outputs resources

# Import DeMark-World
print("Initializing DeMark-World...")
try:
    from demark_world.core import DeMarkWorld
    from demark_world.schemas import CleanerType

    print("✅ DeMark-World imported successfully!")
    print("\nℹ️  Models will be downloaded automatically on first use.")
    print("   This may take a few minutes depending on your connection.")
except Exception as e:
    print(f"❌ Error: {e}")
    import traceback
    traceback.print_exc()

/content/DeMark-World
Initializing DeMark-World...
/content/DeMark-World


2026-02-10 10:51:45.034 | WARNING  | src.demark_world.models.model.modules.feat_prop:<module>:14 - mmcv is not available, using a fallback implementation
2026-02-10 10:51:45.054 | DEBUG    | src.demark_world.utils.devices_utils:get_device:14 - Using device: cuda


/content/DeMark-World


Flax classes are deprecated and will be removed in Diffusers v1.0.0. We recommend migrating to PyTorch classes or pinning your version of Diffusers.
Flax classes are deprecated and will be removed in Diffusers v1.0.0. We recommend migrating to PyTorch classes or pinning your version of Diffusers.
/content/DeMark-World/src/demark_world/iopaint/model/ldm.py:264: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  @torch.cuda.amp.autocast()


✅ DeMark-World imported successfully!

ℹ️  Models will be downloaded automatically on first use.
   This may take a few minutes depending on your connection.


## Step 7: Upload Your Video

Upload any AI-generated video with watermarks (Sora, Runway, Veo, Pika, Kling, Luma, etc.)

In [17]:
from google.colab import files
from pathlib import Path
import shutil

print("Please upload your video file:")
uploaded = files.upload()

# Get the uploaded filename
input_filename = list(uploaded.keys())[0]
input_path = Path(input_filename)

# Move to resources folder
shutil.move(input_filename, f"resources/{input_filename}")
input_video_path = Path(f"resources/{input_filename}")

print(f"\n✅ Uploaded: {input_filename}")
print(f"📁 Saved to: {input_video_path}")

Please upload your video file:


Saving 20260201_2321_01kgd0288sfg1r5807y1p1xs98.mp4 to 20260201_2321_01kgd0288sfg1r5807y1p1xs98.mp4

✅ Uploaded: 20260201_2321_01kgd0288sfg1r5807y1p1xs98.mp4
📁 Saved to: resources/20260201_2321_01kgd0288sfg1r5807y1p1xs98.mp4


## Step 8: Process Video - LAMA (Fast)

**LAMA**: Fast processing, good quality. Recommended for most use cases.

In [18]:
from pathlib import Path
from demark_world.core import DeMarkWorld
from demark_world.schemas import CleanerType

# Set output path
output_video_path = Path("outputs/watermark_removed_lama.mp4")

print("🚀 Processing with LAMA model (fast mode)...")
print("⏱️  Typical speed: 2-5 seconds per second of video (T4 GPU)")
print("")

# Initialize and run
demark = DeMarkWorld(cleaner_type=CleanerType.LAMA)
demark.run(input_video_path, output_video_path)

print(f"\n✅ Done! Video saved to: {output_video_path}")
print("Proceed to Step 10 to download.")

2026-02-10 10:52:32.298 | DEBUG    | src.demark_world.utils.download_utils:ensure_model_downloaded:17 - Downloading model from https://github.com/linkedlist771/DeMark-World/releases/download/V0.0.1/watermark-best.pt


🚀 Processing with LAMA model (fast mode)...
⏱️  Typical speed: 2-5 seconds per second of video (T4 GPU)



Downloading: 100%|██████████| 114M/114M [00:03<00:00, 34.9MB/s]
2026-02-10 10:52:36.116 | SUCCESS  | src.demark_world.utils.download_utils:ensure_model_downloaded:33 - ✓ Model downloaded: /content/DeMark-World/resources/checkpoint/watermark-best.pt
2026-02-10 10:52:36.118 | DEBUG    | src.demark_world.watermark_detector:__init__:20 - Begin to load yolo water mark detet model.
2026-02-10 10:52:36.785 | DEBUG    | src.demark_world.watermark_detector:__init__:24 - Yolo water mark detet model loaded from /content/DeMark-World/resources/checkpoint/watermark-best.pt.
2026-02-10 10:52:36.791 | INFO     | src.demark_world.cleaner.lama_cleaner:__init__:25 - lama not found in /root/.cache, try to downloading
2026-02-10 10:52:36.791 | INFO     | src.demark_world.iopaint.download:cli_download_model:27 - Downloading lama...
Downloading: "https://github.com/Sanster/models/releases/download/add_big_lama/big-lama.pt" to /root/.cache/torch/hub/checkpoints/big-lama.pt
100%|██████████| 196M/196M [00:01<0


✅ Done! Video saved to: outputs/watermark_removed_lama.mp4
Proceed to Step 10 to download.


## Step 9: Process Video - E2FGVI_HQ (High Quality)

**E2FGVI_HQ**: Time-consistent, flicker-free results. Slower but higher quality.

⚠️ **Skip this if you already ran LAMA above!** Only run this if you want higher quality.

In [ ]:
from pathlib import Path
from demark_world.core import DeMarkWorld
from demark_world.schemas import CleanerType

# Set output path
output_video_path = Path("outputs/watermark_removed_e2fgvi.mp4")

print("🚀 Processing with E2FGVI_HQ model (high quality, flicker-free)...")
print("⏱️  Typical speed: 10-20 seconds per second of video (T4 GPU)")
print("⚠️  This will take longer but produces superior results")
print("")

# Initialize and run
demark = DeMarkWorld(cleaner_type=CleanerType.E2FGVI_HQ)
demark.run(input_video_path, output_video_path)

print(f"\n✅ Done! Video saved to: {output_video_path}")
print("Proceed to Step 10 to download.")

## Step 10: Download Processed Video

In [19]:
from google.colab import files

# Download the output video
# Use the appropriate filename based on which method you used
output_file = str(output_video_path)  # Will use the last output_video_path from Step 8 or 9

print(f"Downloading: {output_file}")
files.download(output_file)
print("✅ Download started!")

Downloading: outputs/watermark_removed_lama.mp4


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

✅ Download started!


## Step 11 (Optional): Batch Processing

Process multiple videos at once.

In [ ]:
# Upload multiple videos
from google.colab import files
import shutil
from pathlib import Path

print("Upload multiple video files:")
uploaded_files = files.upload()

# Create batch input/output folders
batch_input = Path("batch_input")
batch_output = Path("batch_output")
batch_input.mkdir(exist_ok=True)
batch_output.mkdir(exist_ok=True)

# Move uploaded files
for filename in uploaded_files.keys():
    shutil.move(filename, batch_input / filename)

print(f"\n✅ {len(uploaded_files)} files uploaded and ready for processing")

In [ ]:
# Process all videos in batch
from pathlib import Path
from demark_world.core import DeMarkWorld
from demark_world.schemas import CleanerType
from tqdm import tqdm

batch_input = Path("batch_input")
batch_output = Path("batch_output")

# Get all video files
video_files = list(batch_input.glob("*.mp4")) + list(batch_input.glob("*.mov")) + list(batch_input.glob("*.avi"))

print(f"Processing {len(video_files)} videos...")
print("")

# Initialize model (LAMA for speed)
demark = DeMarkWorld(cleaner_type=CleanerType.LAMA)

# Process each video
for video_file in tqdm(video_files, desc="Processing videos"):
    output_file = batch_output / f"cleaned_{video_file.name}"
    print(f"\nProcessing: {video_file.name}")

    try:
        demark.run(video_file, output_file)
        print(f"✅ Completed: {output_file.name}")
    except Exception as e:
        print(f"❌ Error processing {video_file.name}: {e}")

print("\n" + "="*60)
print("✅ Batch processing complete!")
print("="*60)

In [ ]:
# Download all processed videos
from google.colab import files
from pathlib import Path

batch_output = Path("batch_output")

print("Downloading all processed videos...")
for video_file in batch_output.glob("*.mp4"):
    print(f"Downloading: {video_file.name}")
    files.download(str(video_file))

print("\n✅ All downloads started!")

## Tips & Performance Notes

### Model Selection:
- **LAMA**: Fast (~2-5 sec/video-sec), good for quick results and testing
- **E2FGVI_HQ**: Slower (~10-20 sec/video-sec) but produces flicker-free, time-consistent results

### Supported Platforms:
DeMark-World works with watermarks from:
- ✅ OpenAI Sora (v1, v2)
- ✅ Google Gemini / Veo (3.1)
- ✅ Runway Gen-3 / Gen-4
- ✅ Pika Labs
- ✅ Kling AI
- ✅ Luma Dream Machine
- ✅ Any AI video generator with consistent watermarks

### GPU Requirements:
- **Minimum**: Google Colab T4 (free tier)
- **Recommended**: T4 or better for faster processing
- **Note**: Enable GPU in Runtime > Change runtime type

### Common Issues:
1. **Out of Memory**: Try LAMA instead of E2FGVI_HQ
2. **Slow Processing**: Ensure GPU is enabled
3. **Model Download Fails**: Check internet connection, models auto-retry

---

**Project Repository**: https://github.com/linkedlist771/DeMark-World

**License**: MIT

**Citation**:
```
@software{demarkworld2025,
  author = {linkedlist771},
  title = {DeMark-World: Universal AI Video Watermark Removal},
  year = {2025},
  url = {https://github.com/linkedlist771/DeMark-World}
}
```